eng/testing/runtimeConfiguration.targets

```diff
@@ -13,7 +13,7 @@
     <RemoteClientAppConfigFile Condition="'$(GenerateAppConfigurationFile)' == 'true'">$(TargetPath).config</RemoteClientAppConfigFile>
   </PropertyGroup>
 
-  <ItemGroup Condition="'$(IsTestProjectOrExecutable)' == 'true'">
+  <ItemGroup Condition="'$(IsTestProjectOrExecutable)' == 'true' and '$(PublishSingleFile)' != 'true'">
```

Comment:
With https://github.com/dotnet/sdk/pull/14488, we will remove this block when we upgrade to a newer SDK. To not accidentally break single file applications, I would suggest to review that PR.